files/default/sql/ddl/updates/3.7.0.sql

```diff
@@ -101,25 +101,199 @@ CREATE TABLE IF NOT EXISTS `hopsworks`.`feature_group_descriptive_statistics` (
+DROP PROCEDURE IF EXISTS PopulateFeatureDescriptiveStatistics;
+DROP PROCEDURE IF EXISTS BatchPopulatePopulateFeatureDescriptiveStatistics;
+DROP PROCEDURE IF EXISTS PopulateTrainingDatasetStatistics;
+DROP PROCEDURE IF EXISTS BatchPopulateTrainingDatasetStatistics;
+DROP PROCEDURE IF EXISTS PopulateFeatureDescriptiveStatisticsForTraingingDataset;
+DROP PROCEDURE IF EXISTS BatchPopulateFeatureDescriptiveStatisticsForTraingingDataset;
+DROP PROCEDURE IF EXISTS UpdateFeatureStoreActivity;
+DROP PROCEDURE IF EXISTS BatchUpdateFeatureStoreActivity;
+DROP PROCEDURE IF EXISTS UpdateFeatureStoreActivityForTrainingDatset;
+DROP PROCEDURE IF EXISTS BatchUpdateFeatureStoreActivityForTrainingDatset;
+
+DELIMITER //
+
+-- Generic batch procedure to run batch jobs calls on the input procedure name 
+CREATE PROCEDURE BatchProc(proc_name VARCHAR(255), batch_size INT, start_id INT, end_id INT)
+BEGIN
+    DECLARE current_id INT;
+    DECLARE done BOOLEAN DEFAULT FALSE;
+
+    SET current_id = start_id;
+
+    WHILE NOT done DO
+        IF current_id + batch_size > end_id THEN
+            SET batch_size = end_id - current_id + 1;
+            SET done = TRUE;
+        END IF;
+
+        SELECT CONCAT('Processing batch from ', current_id, ' to ', current_id + batch_size - 1) AS log_message;
+
+        SET @sql_stmt := CONCAT('CALL ', proc_name, '(', current_id, ',', current_id + batch_size - 1, ')');
+
+        PREPARE stmt FROM @sql_stmt;
+        EXECUTE stmt;
+        DEALLOCATE PREPARE stmt;
+
+        SET current_id = current_id + batch_size;
+    END WHILE;
+
+    SELECT 'All batches processed' AS log_message;
+
+END //
+
+-- Procedure to populate feature_group_statistics
+CREATE PROCEDURE PopulateFeatureGroupStatistics(start_id INT, end_id INT)
+BEGIN
+  -- -- insert new feature_group_statistics. For the same feature group commit statistics, only insert the last computed statistics. For non-time-travel feature groups, insert all computed statistics.
+  INSERT INTO `hopsworks`.`feature_group_statistics` (id, computation_time, feature_group_id, window_end_commit_time)
   WITH sorted_feature_store_statistic AS (
-    SELECT `feature_store_statistic`.*, ROW_NUMBER() OVER (PARTITION BY feature_group_id, IFNULL(feature_group_commit_id, UUID()) ORDER BY commit_time DESC) as rn -- group by fg_id and commit_id, and sort by commit_time --
-    FROM `feature_store_statistic` WHERE feature_group_id IS NOT NULL
+    SELECT `hopsworks`.`feature_store_statistic`.*, ROW_NUMBER() OVER (PARTITION BY feature_group_id, IFNULL(feature_group_commit_id, UUID()) ORDER BY commit_time DESC) as rn
+    FROM `hopsworks`.`feature_store_statistic` WHERE feature_group_id IS NOT NULL and feature_group_id BETWEEN start_id AND end_id
   )
   SELECT id, commit_time as computation_time, feature_group_id, IFNULL(feature_group_commit_id, UNIX_TIMESTAMP(commit_time)*1000) as window_end_commit_time
-  FROM sorted_feature_store_statistic WHERE rn = 1; -- last computed stats --
```

Comment:
Although this SQL statement is run in batches now, the inline comment still provides details of why we do `rn = 1;`. We can edit it to "select last computed stats in the current fg stats batch" if you prefer, but I wouldn't remove it.